content_handler/vulkan_rasterizer.cc

```diff
@@ -445,48 +445,19 @@ bool VulkanRasterizer::Draw(std::unique_ptr<flow::LayerTree> layer_tree) {
     return false;
   }
 
-  compositor_context_.engine_time().SetLapTime(layer_tree->construction_time());
-
-  const SkISize& frame_size = layer_tree->frame_size();
-
-  auto update = mozart::SceneUpdate::New();
-  // TODO(abarth): Support incremental updates.
-  update->clear_resources = true;
-  update->clear_nodes = true;
-
-  if (frame_size.isEmpty()) {
-    update->nodes.insert(mozart::kSceneRootNodeId, mozart::Node::New());
-    // Publish the updated scene contents.
-    // TODO(jeffbrown): We should set the metadata's presentation_time here too.
-    scene_->Update(std::move(update));
-    auto metadata = mozart::SceneMetadata::New();
-    metadata->version = layer_tree->scene_version();
-    scene_->Publish(std::move(metadata));
-    FTL_LOG(ERROR) << "Publishing empty node";
-
-    return false;
-  }
-
   flow::CompositorContext::ScopedFrame frame = compositor_context_.AcquireFrame(
       nullptr, nullptr, true /* instrumentation enabled */);
 
+  // Perform layer tree preroll tasks.
   layer_tree->Preroll(frame);
 
-  flow::SceneUpdateContext context(update.get(), surface_producer_.get());
-  auto root_node = mozart::Node::New();
-  root_node->hit_test_behavior = mozart::HitTestBehavior::New();
-  layer_tree->UpdateScene(context, root_node.get());
-  update->nodes.insert(mozart::kSceneRootNodeId, std::move(root_node));
-
-  // Publish the updated scene contents.
-  // TODO(jeffbrown): We should set the metadata's presentation_time here too.
-  scene_->Update(std::move(update));
-  auto metadata = mozart::SceneMetadata::New();
-  metadata->version = layer_tree->scene_version();
-  scene_->Publish(std::move(metadata));
+  // Publish the layer tree to the session.
+  flow::SceneUpdateContext context(surface_producer_.get());
```

Comment:
We should figure out the hit-testing API ASAP, so that we know if we need to make any significant changes here.  (Mike is starting to look at this)